### PR TITLE
feat(top-app-bar): Add dense

### PR DESCRIPTION
**Pull request by williamernest**
closes: #2426 
Adds the `--dense` style to the top app bar.

### CODE REVIEW

File: packages/mdc-top-app-bar/mdc-top-app-bar.scss [hunk 1]
```diff
@@ -145,5 +145,36 @@
...
+}
+
+// Specific styles for prominent and dense styled top app bar
+.mdc-top-app-bar--dense.mdc-top-app-bar--prominent {
+  .mdc-top-app-bar__row {
+    height: $mdc-top-app-bar-dense-row-height * 2;
+  }
+
+  .mdc-top-app-bar__section {
+    padding: 0 $mdc-top-app-bar-dense-section-horizontal-padding;
```
Comment (moog16) [inline packages/mdc-top-app-bar/mdc-top-app-bar.scss]: I don't think you need this css rule. Isn't it covered by line 154?

File: demos/top-app-bar.html [hunk 2]
```diff
@@ -256,10 +261,16 @@ <h3 class="mdc-typography--title">Demo Controls</h3>
           appBar = mdc.topAppBar.MDCTopAppBar.attachTo(appBarEl);
         });
 
+        denseCheckbox.addEventListener('change', function()  {
+          appBarEl.classList[this.checked ? 'add' : 'remove']('mdc-top-app-bar--dense');
+
+          shortCheckbox.disabled = this.checked || prominentCheckbox.checked;
```
Comment (kfranqueiro) [inline demos/top-app-bar.html]: Dense disables short, but short does not disable dense

File: packages/mdc-top-app-bar/mdc-top-app-bar.scss [hunk 3]
```diff
@@ -145,5 +159,24 @@
...
+  }
+
+  .mdc-top-app-bar__section {
+    padding: 0 $mdc-top-app-bar-section-horizontal-padding;
+  }
+
+  .mdc-top-app-bar__title {
+    @include mdc-rtl-reflexive-box(padding, left, $mdc-top-app-bar-title-left-padding);
+
+    padding-bottom: $mdc-top-app-bar-prominent-dense-title-bottom-padding;
```
Comment (kfranqueiro) [inline packages/mdc-top-app-bar/mdc-top-app-bar.scss]: This is also overriding mobile-specific title bottom padding for prominent (not dense). Is that intentional?

(TBH I find it hard to follow the plethora of different title paddings but I get the impression it's to counteract outer padding...)

Comment (williamernest) [reply packages/mdc-top-app-bar/mdc-top-app-bar.scss]: It's intentional. The `--dense` style is desktop only, so any styles it applies should override any other styles from the mobile media queries.
